### PR TITLE
DOC: add `unique_all` example

### PR DESCRIPTION
**Pull request by luxedo**
Add example for `unique_all` for https://github.com/numpy/numpy/issues/21351

[skip actions][skip azp][skip cirrus]

### CODE REVIEW

**Comment by mattip** [comment]
Doctests are failing. I think a `NORMALIZE_WHITESPACE` directive will allow the nicer formatting you have here instead of cramming everything onto one line.
```
Expected:
    UniqueAllResult(
            values=array([1, 2]),
            indices=array([0, 2]),
            inverse_indices=array([0, 0, 1]),
            counts=array([2, 1]))
Got:
    UniqueAllResult(values=array([1, 2]), indices=array([0, 2]), inverse_indices=array([0, 0, 1]), counts=array([2, 1]))
```

**Comment by luxedo** [comment]
image

It didn't worked. We are not using the vanilla doctest library right? Maybe there's something to do with the `scipy_doctest` plugin? @ev-br

File: numpy/lib/_arraysetops_impl.py [hunk 1]
```diff
@@ -433,6 +433,15 @@ def unique_all(x):
     --------
     unique : Find the unique elements of an array.
 
+    Examples
+    --------
+    >>> np.unique_all([1, 1, 2])  # doctest: +NORMALIZE_WHITESPACE
+    UniqueAllResult(
+            values=array([1, 2]),
```
Comment (mattip) [inline numpy/lib/_arraysetops_impl.py:440]: There is no whitespace between `(` and `values=` so the formatting must be like this
```suggestion
    UniqueAllResult(values=array([1, 2]),
```

Comment (ev-br) [reply numpy/lib/_arraysetops_impl.py:440]: This is likely to just fix itself with scipy_doctest BTW. While numpy is on reguide-check, what Matti said.

Comment (luxedo) [reply numpy/lib/_arraysetops_impl.py]: Thank you guys! Doctests passed

**Comment by mattip** [comment]
Thanks @luxedo

**Pull request merged**